### ISSUE TITLE
Proof fields not used.

### ISSUE THREAD

**Issue by davebryson** [handshake-org/urkel#8]
`proof.isSane()` checks for nil values for the type `deadend`.  Is there even a need for this?   Looking through the code it appears there's no possible way for `key,value,hash` to ever be set for `deadend`.  Is it simply good enough to check for the type without testing the unused fields?  Or are these fields placeholders for future reference?

**Comment by chjj** [comment]
Yeah, the serialization enforces most of the checks in `isSane()` already. A lot of the sanity checks are just there on account of my paranoia.
